File: frontend/interfaces/category.ts
import type { ImageInterface } from '~/interfaces/image'

export interface CategoryInterface {
  id?: number
  title?: string
  description?: string
  image?: ImageInterface
  meta_title?: string
  meta_url?: string
  meta_description?: string
  meta_keywords?: string
  parent?: number
  children?: CategoryInterface[]
  image_upload?: File
  nested_categories?: CategoryInterface[]
}

export interface CategoryPostitonInterface {
  id?: number
  parent_id?: string
  position: string
}

File: backend/voidcms/mdls/Category.py
from django.db import models
from mptt.models import MPTTModel, TreeForeignKey

class Category(MPTTModel):
    id = models.AutoField(primary_key=True)
    image = models.ForeignKey('Image', on_delete=models.SET_NULL, blank=True, null=True)
    title = models.CharField(max_length=100, blank=False)
    description = models.TextField(blank=True)
    #  Positioning
    parent = TreeForeignKey('self', on_delete=models.SET_NULL, blank=True, null=True, related_name='children')
    # Meta
    meta_url = models.CharField(max_length=100, blank=True, unique=True)
    meta_title = models.CharField(max_length=100, blank=True)
    meta_description = models.TextField(blank=True)
    meta_keywords = models.CharField(max_length=255, blank=True)

    class MPTTMeta:
        order_insertion_by = ['id']


    def delete(self, *args, **kwargs):
        if self.image:
            self.image.delete()
        super().delete(*args, **kwargs)
    
    def __str__(self):
        return self.title
